packages/swapper/src/swappers/cow/utils/helpers/helpers.ts

```diff
@@ -0,0 +1,61 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { Asset } from '@shapeshiftoss/types'
+import { AxiosResponse } from 'axios'
+
+import { SwapError, SwapErrorTypes } from '../../../../api'
+import { bn, bnOrZero } from '../../../utils/bignumber'
+import { CowSwapperDeps } from '../../CowSwapper'
+import { CowSwapPriceResponse } from '../../types'
+import { cowService } from '../cowService'
+
+const USDC_CONTRACT_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
+const USDC_ASSET_PRECISION = 6
+
+export const getUsdRate = async (
+  { assetService, apiUrl }: CowSwapperDeps,
+  input: Pick<Asset, 'assetId'>
+): Promise<string> => {
+  const { assetId } = input
+
+  const { assetReference: erc20Address, assetNamespace } = fromAssetId(assetId)
+
+  if (assetNamespace !== 'erc20') {
```

Comment:
this will fail on ETH

Reply:
check for `slip44` too and comment as to why

Reply:
I did this on purpose so it would fail on ETH, because only WETH is supported on CowSwap (ETH needs to be wrapped to WETH before doing anything on CowSwap). Please suggest if I did something wrong, should I return the USD rate of WETH when called with ETH ?

Reply:
ah, in that case we're all good